addons/hack/fakegen.go

```diff
@@ -0,0 +1,12 @@
+package hack
+
+import (
+	crtclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+//go:generate counterfeiter -o ../pkg/fakeclusterclient/crtclusterclient.go --fake-name CRTClusterClient . CrtClient
```

Comment:
If ```counterfeiter``` is going to be used here, I think Makefile needs to have a way to download it if not existing in order for ```generate-fakes``` to work. For example, for ```packageclients```, I used the root ```hack/tools/bin``` for now (otherwise, a ```hack/tools/bin``` can be introduced under addons):

https://github.com/vmware-tanzu/tanzu-framework/blob/main/packageclients/Makefile#L17
https://github.com/vmware-tanzu/tanzu-framework/blob/main/packageclients/Makefile#L12

Reply:
I added hack/tools for addons